Objects/longobject.c

```diff
@@ -514,12 +514,11 @@ PyLong_AsLongAndOverflow(PyObject *vv, int *overflow)
             i = -(i);
         }
```

Comment:
Would it be possible to start at "x = v->ob_digit[0];" and "i=1;" to avoid starting the loop with x=0 which means one useless if at the first iteration.

With 64-bit unsigned long and PyLong_SHIFT, we can even combine two digits without having to check for overflow, no?

Reply:
> we can even combine two digits without having to check for overflow, no?

Something like:

```
/* in the default: case, i >= 2 */
assert(i >= 2);
#if ((ULONG_MAX >> PyLong_SHIFT)) >= ((1UL << PyLong_SHIFT) - 1)
  /* use 2 digits */
  --i;
  x= digit[i];
  x <<=PyLong_SHIFT;
  --i;
  x |= digit[i];
#else
  /* use 1 digit */
  --i
  assert(ULONG_MAX >= ((1UL << PyLong_SHIFT) - 1);
  x= digit[i];
#endif
while (--i >= 0) { ... }
```